lib/Doctrine/Common/Proxy/ProxyGenerator.php

```diff
@@ -537,24 +560,35 @@ public function __set(\$name, \$value)
      * @param \Doctrine\Common\Persistence\Mapping\ClassMetadata $class
      *
      * @return string
+     * @throws \ReflectionException
```

Comment:
Please remove, besides being private API only, this can never happen as we check for `hasMethod()` before calling `getMethod()`.